src/kagglehub/gcs_upload.py

```diff
@@ -66,16 +70,16 @@ def _upload_blob(file_path: str, model_type: str):
         token_exception = "'token' field missing from response"
         raise BackendError(token_exception)
 
+    headers = {"Content-Type": "application/octet-stream"}
+
+    # TODO(312511716): add resumable upload
     with open(file_path, "rb") as f:
-        headers = {"Content-Type": "application/octet-stream"}
-        # TODO(b/312511716): add resumable upload
-        # TODO(b/312511716): Implement a progress bar for the upload using tqdm
-        gcs_response = requests.put(response["createUrl"], data=f, headers=headers, timeout=600, stream=True)
-        if gcs_response.status_code != 200:  # noqa: PLR2004
-            gcs_exception = (
-                f"Failed to upload to GCS. Status: {gcs_response.status_code}, Response: {gcs_response.text}"
-            )
-            raise Exception(gcs_exception)
+        with tqdm(total=file_size, desc="Uploading", unit="B", unit_scale=True, unit_divisor=1024) as pbar:
+            reader_wrapper = CallbackIOWrapper(pbar.update, f, "read")
+            gcs_response = requests.put(response["createUrl"], data=reader_wrapper, headers=headers, timeout=600)
+            if gcs_response.status_code not in [200, 201]:
+                upload_fail_message = f"Upload failed with status code: {gcs_response.status_code}"
```

Comment:
Before, we were including `gcs_response.text` in the error returned to the user. It may contain useful information to let the users fix the issue.